shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -296,14 +289,9 @@ - (void)dealloc {
   for (SemanticsObject* child in _children) {
     child.parent = nil;
   }
-  [_children removeAllObjects];
-  [_children release];
-  [_childrenInHitTestOrder release];
 
   _parent = nil;
```

Comment:
can remove this

Reply:
Per discussion in the previous PR it's not clear we can. Instead maybe we can add a comment for these two lines explaining why they are necessary.

Reply:
> Per discussion in the previous PR

To link it back: https://github.com/flutter/engine/pull/52729#discussion_r1599519065

Reply:
Added a comment.